### DISCUSSION TITLE
Lock acquired with PDOStore is not get released in another session

### DISCUSSION THREAD

**Discussion by t0wqa** [symfony/symfony#42878] [Q&A] [answered]
Hi everyone!

I am using Symfony/Lock 4.4 with PDOStore. 

In console command A, I create a Lock with TTL 3000 using LockFactory::createLock(...) method, and then call acquire() on it. Then, in another command, command B, I need to manually drop this lock before it expires. So I call LockFactory::createLock with same params as in previous command, and then call release() on it.

What I expect: lock is dropped
Actual behaviour: lock remains in database, as if as Lock::release() method was never called.

Does anyone know, what is the reason for the problem?

**Comment by stof** [comment, marked as answer]
Calling `createLock` with the same params will *not* allow you to release it in the command. It will forbid you to acquire it in the command due to the lock being already acquired.

If you want to transfer the lock acquired in some place to release it in a different process, you need to transmit the `\Symfony\Component\Lock\Key` of that lock instead of creating another Key racing for the same resource (and you need to disable the `$autoRelease` feature of the Lock object to avoid releasing it automatically when it is garbage-collected)

**Reply by t0wqa** [reply]
Thank you! Things getting more clear now.

Let me give you a little bit more of context:

We have a long-running command processing entities. The command runs every hour. When one entity gets processed, we lock it so it cannot be processed in the next command executions. The lock TTL depends on another entitiy's (which is related to the original) dateTime field and may be really long, for example, 3 weeks.

But sometimes, we need to manually drop the lock. It is triggered in another place, in API Endpoint. And it may happen, for example, in 3, 5 or even 10 days since the lock was acquired. I understand that we can serialize Key when creating Lock and store it somewhere, but it doesn't seem to be a good solution. What do you think? Maybe we should use some another solution for our long-running locks, not Symfony Lock?

**Reply by stof** [reply]
yeah, this looks like something that should be solved using a business rule in the entity rather than a Lock.
